rsconnect/api.py

```diff
@@ -520,6 +520,46 @@ class ServerDetails(TypedDict):
     python: ServerDetailsPython
 
 
+class ExecutorKwargs(TypedDict):
+    """
+    This describes possible kwargs that can be passed to the RSConnectExecutor.
+    """
+
+    path: NotRequired[str]
+    server: NotRequired[str | None]
+    exclude: NotRequired[tuple[str, ...]]
+    new: NotRequired[bool]
+    app_id: NotRequired[str | None]
+    title: NotRequired[str | None]
+    visibility: NotRequired[str | None]
+    disable_env_management: NotRequired[bool | None]
+    env_vars: NotRequired[dict[str, str]]
+
+
+class ExecutorState(ExecutorKwargs, TypedDict):
```

Comment:
I'm hopeful that these can become normal members of the executor class, and drop the concept of the state dictionary.

Reply:
Done.